src/components/shared/TermsOfUseModal.tsx

```diff
@@ -0,0 +1,155 @@
+import React, { useEffect, useState } from "react";
+import { useTranslation } from "react-i18next";
+import { Field, Formik } from "formik";
+import cn from "classnames";
+import axios from "axios";
+import i18n from "../../i18n/i18n";
+import DOMPurify from "dompurify";
+
+
+const TermsOfUseModal = () => {
+	const { t } = useTranslation();
+	const initialValues = {};
+	const [termsContent, setTermsContent] = useState<string>("");
+	const [agreedToTerms, setAgreedToTerms] = useState(true);
+
+	// Check if already accepted terms
+	useEffect(() => {
+		const checkTerms = async () => {
+		  try {
+			const response = await axios.get("/admin-ng/user-settings/settings.json");
+			// @ts-expect-error TS(7006): Parameter 'result' implicitly has an 'any' type.
+			const isAgreed = response.data.results.some(result => result.key === "agreedToTerms" && result.value === "true");
```

Comment:
I would make this

```suggestion
			const isAgreed = response.data.results.find(result => result.key === "agreedToTerms").value === "true";
```

🤷‍♀️